src/protocols/permissions.ts

```diff
@@ -248,6 +264,70 @@ export class PermissionsProtocol {
     };
   }
 
+  /**
+   * Validates the given Permissions protocol RecordsWrite. It can be a request, grant, or revocation.
+   */
+  public static validateSchema(recordsWriteMessage: RecordsWriteMessage, dataBytes: Uint8Array): void {
+    const dataString = Encoder.bytesToString(dataBytes);
+    const dataObject = JSON.parse(dataString);
+    if (recordsWriteMessage.descriptor.protocolPath === PermissionsProtocol.requestPath) {
+      validateJsonSchema('PermissionRequestData', dataObject);
+    } else if (recordsWriteMessage.descriptor.protocolPath === PermissionsProtocol.grantPath) {
+      validateJsonSchema('PermissionGrantData', dataObject);
+
+      // more nuanced validation that are annoying/difficult to do using JSON schema
+      const permissionGrantData = dataObject as PermissionGrantModel;
+      PermissionsProtocol.validateScope(permissionGrantData.scope);
+      Time.validateTimestamp(permissionGrantData.dateExpires);
+    } else if (recordsWriteMessage.descriptor.protocolPath === PermissionsProtocol.revocationPath) {
+      validateJsonSchema('PermissionRevocationData', dataObject);
+    } else {
+      // defensive programming, should be unreachable externally
+      throw new DwnError(
+        DwnErrorCode.PermissionsProtocolValidateSchemaUnexpectedRecord,
+        `Unexpected permission record: ${recordsWriteMessage.descriptor.protocolPath}`
+      );
+    }
+  }
+
+
+
+  /**
+   * Fetches PermissionsGrantMessage with CID `permissionsGrantId`.
+   * @returns the PermissionsGrantMessage with CID `permissionsGrantId` if message exists
+   * @throws {Error} if PermissionsGrantMessage with CID `permissionsGrantId` does not exist
+   */
```

Comment:
Replace `PermissionsGrantMessage` with `PermissionsGrant` and optionally change the descriptions a bit.

```suggestion
  /**
   * Fetches PermissionsGrant with a `recordId` matching `permissionsGrantId`.
   * @returns the PermissionsGrant with a `recordId` matching `permissionsGrantId` if message exists
   * @throws {Error} if PermissionsGrant with a `recordId` matching `permissionsGrantId` does not exist
   */
```

Reply:
Thanks for picking this up!!